src/main/java/org/littleshoot/proxy/mitm/SubjectAlternativeNameHolder.java

```diff
@@ -58,4 +61,17 @@ private ASN1Encodable parseGeneralName(List<?> nameEntry) {
         }
         throw new IllegalArgumentException(String.valueOf(nameEntry));
     }
+
+    public String getAlternativesIdentifier(String commonName) {
+        if (commonName == null) {
+            throw new IllegalArgumentException(
+                "Error, 'commonName' is not allowed to be null!");
+        }
+        TreeSet<String> alternativesIdentifier = new TreeSet<>();
+        for (ASN1Encodable san : sans) {
+            alternativesIdentifier.add(san.toString());
+        }
+        return alternativesIdentifier.isEmpty() ? commonName : commonName + ALTERNATIVES_IDENTIFIER_DELIMITER +
```

Comment:
> The SAN is not guaranteed to be provided.
> 
> [RFC 6125](https://www.rfc-editor.org/rfc/rfc6125#section-6.4.4), if a SAN exists it should be checked first, otherwise the CN should be used.
> 
> This RFC is also relevant: https://www.rfc-editor.org/rfc/rfc5280#section-4.2.1.6 which also clarifies that this field `may` be absent

@jentfoo the check you mentioned happens here, line 74